e_methods.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Style/CaseEquality
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      result = []
+      my_each { |value| result.push(value) if yield(value) }
+    elsif is_a?(Range)
+      result = []
+      my_each { |value| result.push(value) if yield(value) }
+    else
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) }
+    end
+    result
+  end
+
+  def my_all?(argum = nil)
+    if block_given? && !argum.nil?
```

Comment:
#my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

       expected false
            got true